Use low priority in library analyze queue

poelzi opened the pull request:
The TrackAnalysisScheduler for the library analyze queue uses all CPU cores
which speeds up analysis a lot. Unfortunately those threads fight
heavy with the rest of mixxx when decks are running.
The new LowPriority AnalyzerMode schedules the analyzer thread in QThread::LowPriority
and therefor the other threads more CPU time.
No more skipping audio and super laggy GUI.

poelzi committed:
Use low priority in library analyze queue

The TrackAnalysisScheduler for the library analyze queue uses all CPU cores
which speeds up analysis a lot. Unfortunately those threads fight
heavy with the rest of mixxx when decks are running.
The new LowPriority AnalyzerMode schedules the analyzer thread in QThread::LowPriority
and therefor the other threads more CPU time.
No more skipping audio and super laggy GUI.

uklotzde commented on src/analyzer/trackanalysisscheduler.cpp:
```diff
@@ -63,7 +63,10 @@ TrackAnalysisScheduler::TrackAnalysisScheduler(
                 << numWorkerThreads
-                << "worker threads";
+                << "worker threads. Ideal: "
+                << QThread::idealThreadCount()
```
Mentioning the ideal thread count doesn't matter here after a specific number of worker threads has been requested. Please note that we already start 2 different schedulers with different numbers of worker threads.

Printing the ideal thread count at this point is confusing.

uklotzde commented:
Does this work as intended with the default scheduling strategy on Linux? If not it at least doesn't make it worse.

This PR should target 2.3 instead of master.

poelzi committed:
move priority logic to Worker Thread

The setPriority call needs to be called in the run function,
not the constructor of the Worker class. Add a priority member
to the Worker class and call when initializing.

uklotzde commented on src/util/workerthread.h:
```diff
@@ -134,6 +135,7 @@ class WorkerThread : public QThread {
 
   private:
     const QString m_name;
+    QThread::Priority m_priority;
```
immutable member -> const

uklotzde commented on src/util/workerthread.cpp:
```diff
@@ -19,8 +19,10 @@ std::atomic<int> s_threadCounter(0);
 WorkerThread::WorkerThread(
-        const QString& name)
+        const QString& name,
+        const QThread::Priority priority)
```
wrong signature

poelzi commented:
New pull request for 2.3:
https://github.com/mixxxdj/mixxx/pull/2993

Holzhaus commented:
You don't nee to open a new PR, just rebase and force push to this branch and change the target branch by clicking the edit button next to the PR title.

uklotzde commented:
Please don't open conflicting PRs. I already commented on both, wasted effort and confusing.

uklotzde commented:
Superseded by #2993
